### PR TITLE
Upgrade to react-static 6

### PR DESCRIPTION
**Pull request by NavyAdmiral**
There are a lot of breaking changes in 6 so it's not an easy upgrade. I made the dev-server work, but build is still failing. That is the only thing left to fix in this PR.  

@rupurt Resize package is not working properly anymore so you might want to take a look. I commented it out as it was making the website crash.

Also @rupurt  `react-static` no longer ships with routing capabilities so we need to start using our own. I added `react-router-dom`, but couldn't make it fully work yet as it's still failing production build on exporting html. Should we instead use the `@reach/router` as `react-static` team recommends?

### CODE REVIEW

File: gui/src/Layout.js [hunk 1]
```diff
@@ -1,15 +1,13 @@
 import React from 'react'
-import Routes from 'react-static-routes'
+import { Root } from 'react-static'
```
Comment (tannerlinsley) [inline gui/src/Layout.js]: You'll want to still import Routes here I think?

File: gui/src/Layout.js [hunk 2]
```diff
@@ -52,19 +50,18 @@ const Layout = useHooks(props => {
 
   const {classes, redirectTo} = props
 
-  return (<Router>
-    <Grid container>
+  return (<BrowserRouter>
```
Comment (tannerlinsley) [inline gui/src/Layout.js]: In this case, BrowserRouter would be rendering in the node stage as well. You need to detect which stage the code is in and use the right router and history combo.  Static router for node and browser router for non-node. You can do this by checking for the existence of the Dom. Here is how the old react static worked: 

https://github.com/nozzle/react-static/blob/v5/src/client/components/Router.js#L146

You'll also need to provide the route for the static router. The old react static did it this way: 

https://github.com/nozzle/react-static/blob/v5/src/client/components/Router.js#L180

But you'll likely need to access the staticInfo
To do that. Now that I think of it, there is no API currently to access that directly. I'm going to take a closer look at this in the morning and help you find a solution.

**Comment by rupurt** [comment]
@NavyAdmiral awesome cheers 🍺 

👍 to `@reach/router`. It's from the same guy who built a lot of `react-router` and the API looks fairly similar with a few nice enhancements.

I'll take a deeper look at the PR today. 

Also thanks for the input @tannerlinsley 😄

**Comment by NavyAdmiral** [comment]
> @NavyAdmiral awesome cheers 🍺
> 
> 👍 to `@reach/router`. It's from the same guy who built a lot of `react-router` and the API looks fairly similar with a few nice enhancements.
> 
> I'll take a deeper look at the PR today.
> 
> Also thanks for the input @tannerlinsley 😄

As it stands now the PR should be working. I think I broke few things with babel and I'll fix them later. The only remaining problem is with the `react-resize-detector`dependency that I had to remove.

File: gui/src/Layout.js (hunk 2 above)
Comment (tannerlinsley) [reply gui/src/Layout.js]: @reach/router is actually much easier. It requires much less babysitting for static rendering. You'll be pleasantly surprised.

Comment (NavyAdmiral) [reply gui/src/Layout.js]: The new react-static package works well! Thank you again. 🎉

**Comment by NavyAdmiral** [comment]
@rupurt React-static upgraded its .babelrc so I'm not sure if that started breaking `lint & test`. Regardless, this repo had to upgrade babel deps anyway as they're old & deprecated. I'm not sure why `lint & test` still fail. Other than that, the upgrade to v6 with `react-router-dom` is complete.

**Comment by NavyAdmiral** [comment]
@rupurt Are there any plans to restructure chainlink? Storybook, truffle, gui, geth, flatteners, solidity et al all in 1 package results in cases unrelated upgrades breaking each other like it did now.

**Comment by NavyAdmiral** [comment]
Apparently `standard` errors are unrelated to this PR. They are all over other PRs as well

 
Check https://circleci.com/gh/smartcontractkit/chainlink/14084?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link

**Comment by NavyAdmiral** [comment]
I have no idea as to why truffle is still failing after importing the updated `@babel/register` and `@babel/polyfill` in `truffle.js`. You know what's going on Alex?

**Comment by NavyAdmiral** [comment]
Everything should be okay now. I think the reason Geth is failing on `yarndep` is that react-static-6/webpack-4 export & bundle differently and Circleci might be overusing its resources (?)

Regarding truffle it's a true mystery to me why importing `'babel-register' & 'babel-polyfill'` worked earlier and why do `'@babel/register' & '@babel/polyfill'` no longer work.

**Comment by NavyAdmiral** [comment]
Interesting metrics: build time is _twice_ faster.
Build sizes here:

new
image

old
image

**Comment by NavyAdmiral** [comment]
@rupurt Do you have any idea why just switching to `@babel/register` from `babel-register` and updating the imports correspondingly isn't enough?

**Comment by rupurt** [comment]
@NavyAdmiral the geth tests run the e2e tests. You can replicate this by running `yarn test-e2e` locally. It's failing because the link in the success flash banner is hidden behind the `<Header>` and can't be clicked by the user.

The incorrect positioning is due to the `skipOnMount` prop getting set. I took a bit of a look into why that causes the problem but couldn't find a smoking gun at this point and need to time box this. There is an [open discussion](https://github.com/maslianok/react-resize-detector/issues/55) around something that looks similar to the issue we're running into here.

> Also, standard linting errors are unrelated to this PR. They are all over other PRs as well
> 
> Check gui tests here https://circleci.com/gh/smartcontractkit/chainlink/14084?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link
It looks like the lint task has always been returning a 0 exit code due to snazzy. I've created a new WIP PR to fix it https://github.com/smartcontractkit/chainlink/pull/883

> Finally, are there any plans to restructure chainlink? It's already quite heavy. Storybook, truffle, gui, geth, flatteners, solidity, et al, all in 1 package results in unrelated upgrades breaking each other like it did now.
There are currently no plans to restructure the project but we've discussed it. We previously had multiple `package.json` files but merged them together to get uniform tooling across the project.

**Comment by NavyAdmiral** [comment]
@rupurt The thing is it used to work fine without `skipOnMount` and for some reason it broke with this PR. A wild shot at this would be there are some React 16.7 specific changes that broke it. Wilder shot would be it's because of functional components instead of classes? I'm not quite sure about this because as far as I know both should behave the same unless the class was a `<PureComponent/>` (which it wasn't before we switched to hooks)

**Comment by NavyAdmiral** [comment]
Package react-use-hooks no longer needed with RS6. Waiting on https://github.com/nozzle/react-static/issues/958

**Comment by rupurt** [comment]
Closing in favor of #919

**Pull request closed without merging**